src/main/java/org/minions/devfund/noemiguzman/Squarelotron.java

```diff
@@ -0,0 +1,246 @@
+    /**
+     *
+     * @return current squarelotron
+     */
+    public int[][] getSquarelotron() {
+        return squarelotron.clone();
+    }
+    /**
+     * constructor It fills the 2-dimensional array with the numbers 1 to n squared.
+     *
+     * @param n - the dimension of the squarelotron
+     */
+    public Squarelotron(int n) {
+        this.squarelotron = new int[n][n];
+        this.size = n;
+        for (int i = 0; i < n; i++) {
+            for (int j = 0; j < n; j++) {
+                this.squarelotron[i][j] = i * n + j + 1;
+            }
+        }
+    }
+
+    /**
+     * method to construct a Squarelotron using an array.
+     *
+     * @param array - pass in array
+     * @return - a new Squarelotron class
+     */
+    public static Squarelotron makeSquarelotron(final int[] array) {
+        // test square length
+        int sqrt = (int) Math.sqrt(array.length);
+        if (sqrt * sqrt != array.length) {
+            throw new IllegalArgumentException("bad array provided");
+        }
+        // test the length is between 0 to MAX_SIZE
+        for (int i = 0; i < array.length; i++) {
+            if (array[i] < 0 || array[i] > MAX_SIZE) {
+                throw new IllegalArgumentException("bad array provided");
+            }
+        }
+        // if pass the previous test, we need to construct squarelotron
+        Squarelotron s = new Squarelotron(sqrt);
+        for (int i = 0; i < sqrt; i++) {
+            for (int j = 0; j < sqrt; j++) {
+                s.squarelotron[i][j] = array[i * sqrt + j];
+            }
+        }
+        s.size = sqrt;
+        return s;
+    }
+
+    /**
+     * flip the squarelotron with the indicated ring number.
+     * flip direction - upsideDown
+     *
+     * @param ring - ring number
+     * @return NewSquarelotron
+     */
+
+    public Squarelotron upsideDownFlip(int ring) {
+        // deep copy the board out
+        int[][] deepCopy2 = copySquarelotron();
+
+        // manipulating on deepCopy2, upsideDownFlip on deepCopy2
+        for (int i = 0; i < this.size / DIMENSION; i++) {
+            int[] temp = deepCopy2[i];
+            deepCopy2[i] = deepCopy2[deepCopy2.length - i - 1];
+            deepCopy2[deepCopy2.length - i - 1] = temp;
+        }
+        // get the wanted ring on deepCopy2, and copy those numbers onto deepCopy1
+        // upper bound
+        int[][] deepCopy1 = rotateCorners(ring - 1, this.size - ring, deepCopy2);
+        Squarelotron s = new Squarelotron(this.size);
+        s.squarelotron = deepCopy1;
+        return s;
+    }
+
+    /**
+     * inverse the squarelotron with the indicated ring number.
+     * inverse direction - transpose
+     * @param ring - ring number
+     * @return New Squarelotron
+     */
+    public Squarelotron mainDiagonalFlip(int ring) {
+        // deep copy the board out
+        int[][] deepCopy2 = copyReverseSquarelotron();
+        int[][] deepCopy1 = rotateCorners(ring - 1,this.size - ring, deepCopy2);;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this empty statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)